index.js

```diff
@@ -14,147 +14,31 @@
-              return {testName, result};
-            })
-      );
-    });
-
-    return Promise.all(testResponses).then(results => {
-      if (this.driver_ !== null) {
-        if (typeof this.driver_.browser !== 'undefined') {
-          this.driver_.browser.quit();
-        }
-      }
-      return results;
-    });
-  }
-
-  buildInputsForTest(url, inputs) {
-    let collatedOutputs = {
-      loader: this.loader_
-    };
-    let outputPromises = [];
-    let outputPromise;
-
-    inputs.forEach(input => {
-      input = input.toLowerCase();
-
-      switch (input) {
-        case 'html':
-          outputPromise = new Promise((resolve, reject) => {
-            https.get(url, res => {
-              let body = '';
-              res.on('data', data => {
-                body += data;
-              });
-              res.on('end', () => {
-                collatedOutputs.html = body;
-                resolve();
-              });
-            });
-          });
-          break;
-
-        case 'dom':
-          outputPromise = new Promise((resolve, reject) => {
-            // shut up
-            return resolve();
-            // TODO(paulirish):
-            // https.get(url, (res) => {
-            //   let body = '';
-            //   res.on('data', data => body += data);
-            //   res.on('end', () => {
-            //     try {
-            //       collatedOutputs.dom = DOMParser.parse(body);
-            //     } catch (e){
-            //       reject(e);
-            //     }
-            //     resolve();
-            //   });
-            // });
-          });
-          break;
-
-        case 'chrome':
-          outputPromise = new Promise((resolve, reject) => {
-            if (this.driver_ === null) {
-              this.driver_ = new ChromeDriver();
-            }
-
-            collatedOutputs.driver = this.driver_;
-            resolve();
-          });
-          break;
-
-        case 'url':
-          collatedOutputs.url = url;
-          outputPromise = Promise.resolve();
-          break;
-
-        default:
-          console.warn('Unknown input type: ' + input);
-          break;
-      }
-
-      outputPromises.push(outputPromise);
-    });
-
-    return Promise.all(outputPromises).then(() => collatedOutputs);
-  }
-
-}
-
-TestRunner.get()
-    .then(testRunner => testRunner.test('https://voice-memos.appspot.com/'))
-    .then(results => {
-      console.log(results);
-    }, err => {
-      console.error(err);
-    });
-
-module.exports = {
-  RESPONSE: processor.RESPONSE,
-  ANIMATION: processor.ANIMATION,
-  LOAD: processor.LOAD,
-
-  analyze: function(traceContents, opts) {
-    return processor.analyzeTrace(traceContents, opts);
-  }
-};
+var URL = 'https://voice-memos.appspot.com';
+
+var gatherer = require('./gatherer');
+var auditor = require('./auditor');
+var Browser = require('./browser');
+var _browser;
+
+Browser.construct(URL)
+.then(b => _browser = b)
+.then(gatherer.bind(null, [
```

Comment:
I get what you're going for here, but maybe pull the partial application explicitly into `gatherer` and `auditor` by having them return a function to avoid the `.bind(null` noise?